source/simulator/helper_functions.cc

```diff
@@ -1526,8 +1526,15 @@ namespace aspect
 // explicit instantiation of the functions we implement in this file
 namespace aspect
 {
+#ifndef aspect_simulator_defined
+#define MAYBE_ADVECTION_FIELD(dim) \
+  template struct Simulator<dim>::AdvectionField;
```

Comment:
why is this necessary? what is the error message you are trying to work around? this seems to me like fixing a legitimate problem the wrong way.